src/ioctls/vcpu.rs

```diff
@@ -2256,4 +2336,27 @@ mod tests {
             assert_eq!(vcpu.get_tsc_khz().unwrap(), freq + 500000);
         }
     }
+
+    #[test]
+    #[cfg(target_arch = "aarch64")]
+    fn test_vcpu_attr() {
+        let kvm = Kvm::new().unwrap();
+        let vm = kvm.create_vm().unwrap();
+        let vcpu = vm.create_vcpu(0).unwrap();
+
+        let dist_attr = kvm_bindings::kvm_device_attr {
+            group: KVM_ARM_VCPU_PMU_V3_CTRL,
+            attr: u64::from(KVM_ARM_VCPU_PMU_V3_INIT),
+            addr: 0x0,
+            flags: 0,
+        };
+
+        let mut kvi: kvm_bindings::kvm_vcpu_init = kvm_bindings::kvm_vcpu_init::default();
+        vm.get_preferred_target(&mut kvi)
+            .expect("Cannot get preferred target");
+        kvi.features[0] |= 1 << kvm_bindings::KVM_ARM_VCPU_PSCI_0_2 | 1 << KVM_ARM_VCPU_PMU_V3;
+        assert!(vcpu.vcpu_init(&kvi).is_ok());
+        assert!(vcpu.has_device_attr(&dist_attr).is_ok());
+        assert!(vcpu.set_device_attr(&dist_attr).is_ok());
```

Comment:
Can you add a test for an error case as well, similar to what is done in [test_create_device](https://github.com/rust-vmm/kvm-ioctls/blob/main/src/ioctls/device.rs#L271)?

Reply:
thanks @lauralt -, done for that.